src/main/java/com/configcat/ConfigCatClient.java

```diff
@@ -342,9 +342,29 @@ public void clearDefaultUser() {
 
     @Override
     public void close() throws IOException {
+        closeResources();
+        synchronized (INSTANCE){
+            INSTANCE.remove(sdkKey);
```

Comment:
We should apply the same check that @adams85 mentioned in this [PR](https://github.com/configcat/.net-sdk/pull/50). It can be either an identity check or a `closed` flag.